src/cltl/triple_extraction/__init__.py

```diff
@@ -1,3 +1,11 @@
 """
 The Leolani Triple Extractor Package contains tools to extract SPO triples from Natural Language.
 """
+
+import logging.config
+from pathlib import Path
+
+# create logger
+LOGGING_CONFIG_PATH = Path(__file__).parents[3] / 'config' / 'logging.config'
+logging.config.fileConfig(LOGGING_CONFIG_PATH)
```

Comment:
I would not configure the logger here, i.e. separately for the package, that can be done by the application that uses the package. I think even then you can give the package logger a specific configuration based on the logger name if needed (not sure though, never did that). That means line 13 should be enough here. Then you also don't need the configuration file for the logger in the package. 

And, is there a benefit of having the logger defined here over defining it separately in each file? I'm fine with it though..

Reply:
I think the module hirarchy is already taken care of by `logging.getLogger(__name__)`:
https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial
You don't get seperate child loggers for differnt classes in a module then, if you really want those.

Reply:
I want all logs across our packages (vad, asr, knowledge-extraction, knowledge-representation, language-generation, speech-synthesis) to be formatted the same. I think the formatting we use is something like this:
```
[formatter_pepperFormatter]
format: %(asctime)s - %(levelname)8s - %(name)60s - %(message)s
datefmt: %x %X
```

I thought that to make them equal we could put the same config file across packages. If it is inconvenient/bad practice to add the config files to the package, then I can remove lines 9 and 10 and add something like this

```
formatter = logging.Formatter('%(asctime)s - %(levelname)8s - %(name)60s - %(message)s')
```

to each package. Then we do not need to add the config file to the package. I think though I have to add these lines, right?

```
logger.setLevel(logging.INFO)

handler = logging.StreamHandler(sys.stdout)
handler.setLevel(logging.INFO)
formatter = logging.Formatter('%(asctime)s - %(levelname)8s - %(name)40s - %(message)s')
handler.setFormatter(formatter)
logger.addHandler(handler)
```

Reply:
I still think we shouldn't bother about that on the package level, but on an application level, i.e. put those lines that configure the logger into the "main" function of the application that uses the package instead of the __init__.py . If we configure it inside the __init__.py, it at least makes it harder for the user of the package to configure logging, e.g. if you set to set the log level of the application to debug, you will still get only info logs from the package if we configure the level there as well. Same for the formatting, I think the library user should be able to decide about that.